modules/network/vpc/variables.tf

```diff
@@ -118,44 +122,52 @@ variable "primary_subnetwork" {
   type        = map(string)
   default     = null
   validation {
-    condition     = var.primary_subnetwork == null || can(var.primary_subnetwork["subnet_name"])
+    condition     = var.primary_subnetwork == null ? true : can(var.primary_subnetwork["subnet_name"])
```

Comment:
Why this change? Aren't both of those lines are semantically identical?

Reply:
Unfortunately, HCL doesn't use short-circuiting in conditionals: https://github.com/hashicorp/terraform/issues/24128#issuecomment-1155241478

However, it does for ternary statements (i.e. `<condition> ? <true_action> : <false_action>`).

Reply:
How does lack of short-circuiting changes anything here?
IIUC `can(expr)` will return `false` if `expr` throws an error, so `var.primary_subnetwork == null` shouldn't be a problem here.

Reply:
That's a good point, I'll revert the changes which involve `can()`.